Tools/cases_generator/optimizer_generator.py

```diff
@@ -75,6 +80,10 @@ def type_name(var: StackItem) -> str:
         return "JitOptRef *"
     return "JitOptRef "
 
+def stackref_type_name(var: StackItem) -> str:
+    if var.is_array():
```

Comment:
```suggestion
    assert not var.is_array(), "Unsafe to convert a symbol to an array-like StackRef."
```
Check this in the caller if that makes sense, and raise an analysis error there so we aren't relying on `assert` to handle input errors.